application/comit_node/src/swap_protocols/rfc003/actions/alice/btc_erc20.rs

```diff
@@ -49,24 +49,28 @@ impl OngoingSwap<Alice<Bitcoin, Ethereum, BitcoinQuantity, Erc20Quantity>> {
     }
 }
 
-type AliceActionKind =
-    ActionKind<(), (), (), bitcoin::SendToAddress, ethereum::SendTransaction, bitcoin::SpendOutput>;
-
 impl Actions for SwapStates<Alice<Bitcoin, Ethereum, BitcoinQuantity, Erc20Quantity>> {
-    type ActionKind = AliceActionKind;
+    type ActionKind = alice::ActionKind<
+        (),
+        bitcoin::SendToAddress,
+        ethereum::SendTransaction,
+        bitcoin::SpendOutput,
+    >;
 
-    fn actions(&self) -> Vec<AliceActionKind> {
+    fn actions(&self) -> Vec<Self::ActionKind> {
         use self::SwapStates as SS;
         match *self {
-            SS::Accepted(Accepted { ref swap, .. }) => vec![ActionKind::Fund(swap.fund_action())],
+            SS::Accepted(Accepted { ref swap, .. }) => {
+                vec![alice::ActionKind::Fund(swap.fund_action())]
+            }
             SS::BothFunded(BothFunded {
                 ref alpha_htlc_location,
                 ref beta_htlc_location,
                 ref swap,
                 ..
             }) => vec![
-                ActionKind::Redeem(swap.redeem_action(*beta_htlc_location)),
-                ActionKind::Refund(swap.refund_action(*alpha_htlc_location)),
+                alice::ActionKind::Redeem(swap.redeem_action(*beta_htlc_location)),
```

Comment:
🙃 Best to be homogeneous and use `Self` everywhere

Reply:
This is a known bug, but this PR should fix it: https://github.com/rust-lang/rust/pull/56225